packages/solid-store/src/index.tsx

```diff
@@ -0,0 +1,30 @@
+import { AnyUpdater, shallow, Store } from '@tanstack/store'
+import { Accessor, createSignal, onMount } from 'solid-js'
+
+export type NoInfer<T> = [T][T extends any ? 0 : never]
+
+export function useStore<
+  TState,
+  TSelected = NoInfer<TState>,
+  TUpdater extends AnyUpdater = AnyUpdater,
+>(
+  routerStore: Store<TState, TUpdater>,
+  selector: (state: NoInfer<TState>) => TSelected = (d) => d as any,
+  compareShallow?: boolean,
+): Accessor<TSelected> {
+  const [state, setState] = createSignal<TSelected>(
+    selector(routerStore.state),
+    {
+      equals: compareShallow ? false : shallow,
+    },
+  )
```

Comment:
This will likely need to turn into a solid store and uses solid-store's reconcile to make it granularly reactive.